cd src && diff -rwu gnome-terminal-3.28.2-orig gnome-terminal-3.28.2; [ $? -eq 1 ]
diff -rwu gnome-terminal-3.28.2-orig/src/terminal-util.c gnome-terminal-3.28.2/src/terminal-util.c
--- gnome-terminal-3.28.2-orig/src/terminal-util.c	2019-05-13 19:04:21.562955256 +0300
+++ gnome-terminal-3.28.2/src/terminal-util.c	2019-05-13 19:04:21.570955898 +0300
@@ -111,7 +111,7 @@
     }
 }
 
-static gboolean
+gboolean
 open_url (GtkWindow *parent,
           const char *uri,
           guint32 user_time,
diff -rwu gnome-terminal-3.28.2-orig/src/terminal-util.h gnome-terminal-3.28.2/src/terminal-util.h
--- gnome-terminal-3.28.2-orig/src/terminal-util.h	2019-05-13 19:04:21.562955256 +0300
+++ gnome-terminal-3.28.2/src/terminal-util.h	2019-05-13 19:04:21.570955898 +0300
@@ -110,6 +110,12 @@
 void terminal_util_save_print_settings (GtkPrintSettings *settings,
                                         GtkPageSetup *page_setup);
 
+gboolean
+open_url (GtkWindow *parent,
+          const char *uri,
+          guint32 user_time,
+          GError **error);
+
 G_END_DECLS
 
 #endif /* TERMINAL_UTIL_H */
diff -rwu gnome-terminal-3.28.2-orig/src/terminal-window.c gnome-terminal-3.28.2/src/terminal-window.c
--- gnome-terminal-3.28.2-orig/src/terminal-window.c	2019-05-13 19:04:21.562955256 +0300
+++ gnome-terminal-3.28.2/src/terminal-window.c	2019-05-13 19:24:11.553855074 +0300
@@ -654,6 +654,29 @@
   vte_terminal_copy_clipboard_format (VTE_TERMINAL (priv->active_screen), format);
 }
 
+static void
+action_search_cb (GSimpleAction *action,
+                GVariant *parameter,
+                gpointer user_data)
+{
+  TerminalWindow *window = user_data;
+  TerminalWindowPrivate *priv = window->priv;
+  const char *format_str;
+  VteFormat format;
+  gchar *url;
+  GtkClipboard *clipboard;
+
+  if (priv->active_screen == NULL)
+    return;
+
+  g_assert_nonnull (parameter);
+  g_variant_get (parameter, "&s", &format_str);
+  clipboard = gtk_widget_get_clipboard (GTK_WIDGET (window), GDK_SELECTION_PRIMARY);
+  url = g_strdup_printf ("https://duckduckgo.com/?q=%s", gtk_clipboard_wait_for_text(clipboard));
+  open_url (GTK_WINDOW (window), url, gtk_get_current_event_time (), NULL);
+  g_free(url);
+}
+
 /* Clipboard helpers */
 
 typedef struct {
@@ -1781,6 +1804,7 @@
   g_menu_append (section4, _("_Copy"), "win.copy::text");
   g_menu_append (section4, _("Copy as _HTML"), "win.copy::html");
   g_menu_append (section4, _("_Paste"), "win.paste-text");
+  g_menu_append (section4, _("Search on DuckDuckGo"), "win.search-internet::text");
   if (g_action_get_enabled (G_ACTION (lookup_action (window, "paste-uris"))))
     g_menu_append (section4, _("Paste as _Filenames"), "win.paste-uris");
 
@@ -2119,6 +2143,7 @@
     { "paste-uris",          action_paste_uris_cb,       NULL,   NULL, NULL },
     { "reset",               action_reset_cb,            "b",    NULL, NULL },
     { "select-all",          action_select_all_cb,       NULL,   NULL, NULL },
+    { "search-internet",     action_search_cb,           "s",    NULL, NULL },
     { "size-to",             action_size_to_cb,          "(uu)", NULL, NULL },
     { "tab-detach",          action_tab_detach_cb,       NULL,   NULL, NULL },
     { "tab-move-left",       action_tab_move_left_cb,    NULL,   NULL, NULL },
